nucleus/hk2-config-generator/src/main/java/org/jvnet/hk2/config/ConfigParser.java

```diff
@@ -140,7 +142,9 @@ protected Dom handleElement(XMLStreamReader in,DomDocument document, Dom parent)
         ConfigModel model = document.getModelByElementName(in.getLocalName());
         if(model==null) {
             String localName = in.getLocalName();
-            Logger.getAnonymousLogger().severe("Ignoring unrecognized element "+in.getLocalName() + " at " + in.getLocation());
+            logger.log(Level.WARNING, () -> "Ignoring unrecognized element " + in.getLocalName() + " at line #" + in.getLocation().getLineNumber());
+            logger.log(Level.FINE, () -> "Unrecognized element " + in.getLocalName() + " at "
+                    + in.getLocation().toString().replace("\n", ", "));
```

Comment:
I am not sure if it is good to manipulate the location toString here, but ... or I will say it. It is wrong, but I have found that `EventLocation` (`jaxb-osgi`) really prints `\n` character after every value ... perhaps we could create a PR for them ;-)